### PR TITLE
CP leaderboard implementation

### PR DESCRIPTION
**Pull request by swayam-agrahari**
- Added scheduled tasks to update CP ratings and insert attendance records daily at midnight.
- Implemented API calls to fetch ratings from Codeforces and LeetCode.
- Updated update_cp_ratings function to handle dynamic ratings based on platform.
- Created placeholders for test configuration to avoid hardcoding sensitive data.
- Added tests to validate rating updates for members with Codeforces and LeetCode profiles.

### CODE REVIEW

File: src/db/member.rs [hunk 1]
```diff
@@ -14,4 +13,7 @@ pub struct Member {
     pub sex: String,
     pub year: i32,
     pub macaddress: String,
+    pub leaderboard_id: String,
```
[the remark this answers is not in the document]
Comment (swayam-agrahari) [reply src/db/member.rs:16]: this is the username for their respective cp platform

[the remark this answers is not in the document]
Comment (Wreck-X) [reply src/db/member.rs:16]: ah gotcha, but they should be free to do both like i said not just 1 platform

File: src/main.rs [hunk 2]
```diff
@@ -86,32 +96,174 @@ async fn scheduled_task(pool: Arc<PgPool>) {
...
+    }
+
+    // Test for update_cp_ratings
+    #[tokio::test]
+    async fn test_update_cp_ratings() {
+        let database_url = ""; // Add your database URL here
+        let pool = Arc::new(PgPool::connect(database_url).await.unwrap());
+        update_cp_ratings(pool).await;
+    }
+}
```
[the remark this answers is not in the document]
Comment (swayam-agrahari) [reply src/main.rs:269]: Yeah sure will do it. I also thought about creating more tests for seeing members for dev purpose so this function can lie there.

File: src/main.rs [hunk 3]
```diff
@@ -86,32 +96,174 @@ async fn scheduled_task(pool: Arc<PgPool>) {
...
+                    match update_result {
+                        Ok(_) => println!("Updated rating for {}: {}", member.name, rating),
+                        Err(e) => eprintln!("Failed to update rating for {}: {:?}", member.name, e),
+                    }
+                }
+            }
+        }
+        Err(e) => eprintln!("Failed to fetch members: {:?}", e),
+    }
+}
```
[the remark this answers is not in the document]
Comment (swayam-agrahari) [reply src/main.rs:225]: yes done : )

File: src/db/member.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (swayam-agrahari) [reply src/db/member.rs:16]: alright so to implement this, we might need to create another leaderboard struct as this is one to many relation with foreign key as the member's id (primary key)

[the remark this answers is not in the document]
Comment (Wreck-X) [reply src/db/member.rs:16]: Yep that's right

**Pull request closed without merging**